x-pack/legacy/plugins/epm/server/lib/template/template.test.ts

```diff
@@ -5,9 +5,38 @@
  */
 
 import { getTemplate } from './template';
+import { readFileSync, writeFileSync } from 'fs';
+import path from 'path';
+import { safeLoad } from 'js-yaml';
+import { Field, ProcessFields } from '../field';
+import { generateMappings } from './template';
 
 test('get template', () => {
   const pattern = 'logs-nginx-access-abcd-*';
   const template = getTemplate(pattern);
   expect(template.index_patterns).toStrictEqual([pattern]);
 });
+
+test('tests loading fields.yml', () => {
+  // Load fields.yml file
+  const fieldsYML = readFileSync(path.join(__dirname, '../tests/fields/base.yml'), 'utf-8');
+  const data = safeLoad(fieldsYML.toString());
+  const fields = data as Field[];
+
+  ProcessFields(fields);
+  const mappings = generateMappings(fields);
+  const template = getTemplate('foo', mappings);
+
+  const json = JSON.stringify(template, null, 2);
+  const generatedFile = path.join(__dirname, '../tests/fields/base.template.generate.json');
+
+  // Regenerate the file if `-generate` flag is used
+  if (process.argv.includes('-generate')) {
```

Comment:
`--generate`? UNIX convention is single dash for short arguments (`-g`) and double for long names (`--generate`)